tasks/get_current_archive_list/get_current_archive_list.py

```diff
@@ -0,0 +1,532 @@
+    with open("schemas/input.json", "r") as raw_schema:
+        _INPUT_VALIDATE = fastjsonschema.compile(json.loads(raw_schema.read()))
+    with open("schemas/output.json", "r") as raw_schema:
+        _OUTPUT_VALIDATE = fastjsonschema.compile(json.loads(raw_schema.read()))
+except Exception as ex:
+    # Can't use f"" because of '{}' bug in CumulusLogger.
+    LOGGER.error("Could not build schema validator: {ex}", ex=ex)
+    raise
+
+
+def task(
+    record: Dict[str, Any],
+    s3_access_key: str,
+    s3_secret_key: str,
+    db_connect_info: Dict,
+) -> Dict[str, Any]:
+    """
+    Reads the record to find the location of manifest.json, then uses that information to spawn of business logic
+    for pulling manifest's data into sql.
+
+    Args:
+        record: See input.json for details.
+        s3_access_key: The access key that, when paired with s3_secret_key, allows postgres to access s3.
+        s3_secret_key: The secret key that, when paired with s3_access_key, allows postgres to access s3.
+        db_connect_info: See shared_db.py's get_configuration for further details.
+
+    Returns: See output.json for details.
+    """
+    # Filter out non-manifest files
+    filename = os.path.basename(record[RECORD_S3_KEY][S3_OBJECT_KEY][OBJECT_KEY_KEY])
+
+    if filename != "manifest.json":
+        LOGGER.error(f"Illegal file '{filename}'. Must be 'manifest.json'")
+        raise Exception(f"Illegal file '{filename}'. Must be 'manifest.json'")
+
+    # See https://docs.aws.amazon.com/AmazonS3/latest/userguide/storage-inventory-location.html for json example.
+    manifest = get_manifest(
+        record[RECORD_S3_KEY][S3_OBJECT_KEY][OBJECT_KEY_KEY],
+        record[RECORD_S3_KEY][S3_BUCKET_KEY][BUCKET_NAME_KEY],
+        record[RECORD_AWS_REGION_KEY],
+    )
+
+    user_engine = shared_db.get_user_connection(db_connect_info)
+    admin_engine = shared_db.get_admin_connection(
+        db_connect_info, db_connect_info["user_database"]
+    )
+    # Create initial job
+    # noinspection PyArgumentList
+    job_id = create_job(
+        manifest[MANIFEST_SOURCE_BUCKET_KEY],
+        datetime.utcfromtimestamp(
+            int(manifest[MANIFEST_CREATION_TIMESTAMP_KEY]) / 1000
+        ).replace(tzinfo=timezone.utc),
+        user_engine,
+    )
+
+    try:
+        # noinspection PyArgumentList
+        truncate_s3_partition(manifest[MANIFEST_SOURCE_BUCKET_KEY], admin_engine)
+        # noinspection PyArgumentList
+        update_job_with_s3_inventory_in_postgres(
+            s3_access_key,
+            s3_secret_key,
+            manifest[MANIFEST_SOURCE_BUCKET_KEY],
+            record[RECORD_S3_KEY][S3_BUCKET_KEY][BUCKET_NAME_KEY],
+            record[RECORD_AWS_REGION_KEY],
+            # There will probably only be one file, but AWS leaves the option open.
+            [file[FILES_KEY_KEY] for file in manifest[MANIFEST_FILES_KEY]],
+            manifest[MANIFEST_FILE_SCHEMA_KEY],
+            job_id,
+            admin_engine,
```

Comment:
Shouldn't we use the app user credentials and not the admin credentials here?

Reply:
`permission denied for schema aws_s3`. I'd easily believe it's something to do with how I set the DB up.

Reply:
I just thought of something, we can sure run some queries and see but, it is possible it is locked down to admin on account of the COPY command to prevent a user from accessing an external resource.